### ISSUE TITLE
Signing an IPA with a watch extension

### ISSUE THREAD

**Issue by dkit** [DanTheMan827/ios-app-signer#6]
When I try to install it I get "The app contains a WatchKit app with an invalid bundle identifier. The bundle identifier of a WatchKit app must have a prefix consisting of the companion app's bundle identifier, followed by a '.'."

Is it possible to automatically change a watchkit's bundle id as well?

**Comment by thimohagen** [comment]
Same problem. Any idea how I can fix this?

**Comment by mjnguyen** [comment]
any update on this?  I have a topshelf extension with my tvOS app and have a similar error message when I try to upload to the store

**Comment by dkit** [comment]
I ended up having to unpack the ipa and manually edit the watchkit's plist file and match the bundle id to the one modified by this code.

**Comment by thimohagen** [comment]
How'd you do this? Does this sign them together and have a working watchkit extension?

**Comment by dkit** [comment]
Yes an IPA is just a zip file. So, unzip the IPA, change the bundle id in the watchkit's directory. Zip it back up, and then use this code to sign it.

**Comment by DanTheMan827** [comment]
Zipping the app back up into an IPA file isn't even required

Just use the .app bundle located inside of the Payload folder

The app accepts any .app bundle as input as well as deb, ipa or xcarchive
